src/Transports.AspNetCore.NewtonsoftJson/GraphQLRequestDeserializer.cs

```diff
@@ -0,0 +1,70 @@
+using GraphQL.Server.Transports.AspNetCore.Common;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using GraphQLRequestBase = GraphQL.Server.Common.GraphQLRequest;
+
+namespace GraphQL.Server.Transports.AspNetCore.NewtonsoftJson
+{
+    /// <summary>
+    /// Implementation of an <see cref="IGraphQLRequestDeserializer"/> that uses Newtonsoft.Json.
+    /// </summary>
+    public class GraphQLRequestDeserializer : IGraphQLRequestDeserializer
+    {
+        private readonly JsonSerializer _serializer;
+
+        public GraphQLRequestDeserializer(Action<JsonSerializerSettings> configure)
+        {
+            var settings = new JsonSerializerSettings();
+            configure?.Invoke(settings);
+            _serializer = JsonSerializer.Create(settings); // it's thread safe https://stackoverflow.com/questions/36186276/is-the-json-net-jsonserializer-threadsafe
+        }
+
+        public Task<GraphQLRequestDeserializationResult> DeserializeFromJsonBodyAsync(HttpRequest httpRequest, CancellationToken cancellationToken = default)
```

Comment:
`cancellationToken` is not used in this method. Let's use it - `cancellationToken.ThrowIfCancellationRequested()` in the beginning or after `Peek()` will be enough.

Reply:
lol I'd been thinking about this but forgotten. Nice one, done.